core/pva/src/main/java/org/epics/pva/data/PVABoolArray.java

```diff
@@ -91,9 +91,7 @@ public void encodeType(ByteBuffer buffer, BitSet described) throws Exception
     public void decode(final PVATypeRegistry types, final ByteBuffer buffer) throws Exception
     {
         final int size = PVASize.decodeSize(buffer);
-        boolean[] new_value = value;
-        if (new_value == null  ||  new_value.length != size)
-            new_value = new boolean[size];
+        boolean[] new_value = new boolean[size];
```

Comment:
Can you make them `final` as in the double example?

Reply:
Yes... makes sense.

Reply:
Thanks!